lineedit/display.c

```diff
@@ -705,11 +705,16 @@ void update_editline(void)
 	    = lebuf.pos.line * lebuf.maxcolumn + lebuf.pos.column;
 	if (index == le_main_buffer.length)
 	    break;
-	if (styler_active && index >= le_main_length)
-	    lebuf_print_sgr0(), styler_active = false;
+	if (styler_active && index >= le_main_length) {
+	    lebuf_print_sgr0();
+	    lebuf_print_prompt(prompt.predict);
+	    styler_active = false;
+	}
 	lebuf_putwchar(c, true);
 	index++;
     }
+    lebuf_print_sgr0();
```

Comment:
We should set `styler_active` to false whenever printing sgr0, so that the shell can correctly decide whether it needs to print the styler string again before the next update.
```suggestion
    lebuf_print_sgr0(), styler_active = false;
```